lib/events/athena/integration_test.go

```diff
@@ -0,0 +1,391 @@
+	})
+
+	t.Logf("Initialized Athena test suite %q\n", testID)
+
+	return ac
+}
+
+type infraOutputs struct {
+	topicARN string
+	queueURL string
+}
+
+func (ac *athenaContext) setupInfra(t *testing.T, ctx context.Context) {
+	t.Run("create SNS and SQS and set subscriber with permissions", func(t *testing.T) {
+		snsClient := sns.NewFromConfig(ac.awsCfg)
+		topicCreated, err := snsClient.CreateTopic(ctx, &sns.CreateTopicInput{
+			Name: aws.String(ac.testID),
+		})
+		require.NoError(t, err)
+
+		sqsClient := sqs.NewFromConfig(ac.awsCfg)
+		queueCreated, err := sqsClient.CreateQueue(ctx, &sqs.CreateQueueInput{
+			QueueName: aws.String(ac.testID),
+		})
+		require.NoError(t, err)
+
+		ac.infraOutputs = &infraOutputs{
+			topicARN: aws.ToString(topicCreated.TopicArn),
+			queueURL: aws.ToString(queueCreated.QueueUrl),
+		}
+
+		queueAttr, err := sqsClient.GetQueueAttributes(ctx, &sqs.GetQueueAttributesInput{
+			QueueUrl:       queueCreated.QueueUrl,
+			AttributeNames: []sqsTypes.QueueAttributeName{sqsTypes.QueueAttributeNameQueueArn},
+		})
+		require.NoError(t, err)
+		queueArn := queueAttr.Attributes["QueueArn"]
+
+		type StatementEntry struct {
+			Effect    string
+			Action    []string
+			Resource  string
+			Principal map[string]string
+			Condition map[string]map[string]string
+		}
+		type PolicyDocument struct {
+			Version   string
+			Statement []StatementEntry
+		}
+		sqsAccessPolicy := PolicyDocument{
+			Version: "2012-10-17",
+			Statement: []StatementEntry{
+				{
+					Effect:   "Allow",
+					Action:   []string{"SQS:SendMessage"},
+					Resource: queueArn,
+					Principal: map[string]string{
+						"AWS": "*",
+					},
+					Condition: map[string]map[string]string{
+						"ArnLike": {
+							"aws:SourceArn": *topicCreated.TopicArn,
+						},
+					},
+				},
+			},
+		}
+		marshaledPolicy, err := json.Marshal(sqsAccessPolicy)
+		require.NoError(t, err)
+		_, err = sqsClient.SetQueueAttributes(ctx, &sqs.SetQueueAttributesInput{
+			Attributes: map[string]string{
+				"Policy": string(marshaledPolicy),
+			},
+			QueueUrl: queueCreated.QueueUrl,
+		})
+		require.NoError(t, err)
+
+		_, err = snsClient.Subscribe(ctx, &sns.SubscribeInput{
+			TopicArn: topicCreated.TopicArn,
+			Protocol: aws.String("sqs"),
+			Attributes: map[string]string{
+				"RawMessageDelivery": "true",
+			},
+			Endpoint: aws.String(queueArn),
+		})
+		require.NoError(t, err)
+	})
+
+	t.Run("create glue db if not exists", func(t *testing.T) {
+		glueClient := glue.NewFromConfig(ac.awsCfg)
+		_, err := glueClient.GetDatabase(ctx, &glue.GetDatabaseInput{
+			Name: aws.String(ac.database),
+		})
+		if err != nil {
+			var notFound *glueTypes.EntityNotFoundException
+			if errors.As(err, &notFound) {
+				_, createErr := glueClient.CreateDatabase(ctx, &glue.CreateDatabaseInput{
+					DatabaseInput: &glueTypes.DatabaseInput{
+						Name: aws.String(ac.database),
+					},
+				})
+				require.NoError(t, createErr)
+			} else {
+				require.NoError(t, err)
```

Comment:
I think that this condition is always false. 
This assertion is called inside the context of `if err != nil { require.NoError(t, err)}`.

Reply:
I have used now assert.Fail: https://github.com/gravitational/teleport/pull/25248/commits/584f45e117d38dd5936c89069e8a1850b1045ed0